src/main/java/com/boomi/flow/services/boomi/mdh/match/MatchEntityRepository.java

```diff
@@ -51,21 +54,35 @@ public MatchEntityRepository(MdhClient mdhClient) {
                     .map(entity -> createUpdateEntity(universe, entity))
                     .collect(Collectors.toList());
 
-            // Now we ned to load the match entity information
-            var updateRequest = new BatchUpdateRequest()
-                    .setSource(sourceId)
-                    .setEntities(entities);
+            resultsList.add(getResults(configuration, entities, sourceId, universe));
+        }
+
+        return resultsList.stream()
+                .flatMap(Collection::stream)
+                .filter(result -> result.getEntity() != null && result.getEntity().get(universe.getName()) != null)
+                .map(matchResult -> MatchResponseMapper.createMobjectFromResult(universeId, universe, matchResult))
+                .collect(Collectors.toList());
+    }
 
-            MatchEntityResponse matchResponse = client.queryMatchEntity(configuration.getAtomHostname(),
-                    configuration.getAtomUsername(), configuration.getAtomPassword(), universe.getId().toString(),
-                    updateRequest);
+    private List<MatchEntityResponse.MatchResult> getResults(ApplicationConfiguration configuration,
+                                                       List<BatchUpdateRequest.Entity> entities, String sourceId,
+                                                       Universe universe) {
 
-            List<MatchEntityResponse.MatchResult> matchesResult = matchResponse.getMatchResults();
-            matchesResult.forEach(result -> result.setIdResource(sourceId));
-            results.addAll(matchesResult);
-        }
+        // Now we ned to load the match entity information
+        var updateRequest = new BatchUpdateRequest()
+                .setSource(sourceId)
+                .setEntities(entities);
+
+        MatchEntityResponse matchResponse = client.queryMatchEntity(configuration.getHubHostname(),
+                configuration.getHubUsername(), configuration.getHubToken(), universe.getId().toString(),
+                updateRequest);
+
+        return matchResponse.getMatchResults().stream()
```

Comment:
This should just be a normal `foreach` rather than a stream, as it's not really doing any immutable stream transformations and is just more noisy as a stream